cmd/lncli/commands.go

```diff
@@ -1822,13 +1829,22 @@ func listChainTxns(ctx *cli.Context) error {
 
 	req := &lnrpc.GetTransactionsRequest{}
 
+	start_height := ctx.Int64("start_height")
+	end_height := ctx.Int64("end_height")
+
+	if (start_height > end_height) &&
```

Comment:
This logic looks like a good candidate for a switch statement. What do you think?

```
switch {
case ctx.IsSet("start_height") && ctx.IsSet("end_height"):
        start_height := ctx.Int64("start_height")
        end_height := ctx.Int64("end_height")
        if end_height != -1 && start_height > end_height {
                return errors.New("start_height should be less than or equal to end_height")
        }
        
        req.StartHeight = int32(start_height)
        req.EndHeight = int32(end_height)
        
case ctx.IsSet("start_height"):
        req.StartHeight = int32(ctx.Int64("start_height"))

case ctx.IsSet("end_height"):
        req.EndHeight = int32(ctx.Int64("end_height"))
}
```


It is really important that this check is replicated in the server side. If you follow the full flow the btcwallet library used for checking the range will return the result reversed if  `start_height > end_height` starting from end to start. Because this is something that we do not want to expose in the API (we have the `--reverse` flag for that) we need to catch it before processing the range.

Reply:
> This logic looks like a good candidate for a switch statement. What do you think?
> 
> ```
> switch {
> case ctx.IsSet("start_height") && ctx.IsSet("end_height"):
>         start_height := ctx.Int64("start_height")
>         end_height := ctx.Int64("end_height")
> 
>         if end_height == -1 && start_height != 0 {
>                 return errors.New("...")
>         }
>         
>         if end_height != -1 &&  start_height  > end_height {
>                 return errors.New("...")
>         }
>       
>         req.StartHeight = int32(start_height)
>         req.EndHeight = int32(end_height)
>         
> case ctx.IsSet("start_height"):
>         req.StartHeight = int32(ctx.Int64("start_height"))
> 
> case ctx.IsSet("end_height"):
>         req.EndHeight = int32(ctx.Int64("end_height"))
> }
> ```
> 
> It is really important that this check is replicated in the server side. If you follow the full flow the btcwallet library used for checking the range will return the result reversed if `start_height > end_height` starting from end to start. Because this is something that we do not want to expose in the API (we have the `--reverse` flag for that) we need to catch it before processing the range.

Thank you so much. I would work with the sith statement but are you saying that I should make some changes in the btcwallet as well or write you statement below as comments